test/unit/active_record_batch_enumerator_test.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module JobIteration
+  class ActiveRecordBatchEnumeratorTest < IterationUnitTest
+    SQL_TIME_FORMAT = "%Y-%m-%d %H:%M:%S.%N"
+    test "#each yields batches as relation with the last record's cursor position" do
+      enum = build_enumerator
+      product_batches = Product.order(:id).take(4).in_groups_of(2).map { |product| [product, product.last.id] }
+
+      enum.first(2).each_with_index do |(batch, cursor), index|
+        assert batch.is_a?(ActiveRecord::Relation)
+        assert_equal product_batches[index].first, batch
+        assert_equal product_batches[index].last, cursor
+      end
+    end
+
+    test "#each yields unloaded relations" do
+      enum = build_enumerator
+      relation, _ = enum.first
+
+      refute_predicate relation, :loaded?
+    end
+
+    test "#each doesn't yield anything if the relation is empty" do
+      enum = build_enumerator(relation: Product.none)
+
+      assert_equal([], enum.to_a)
+    end
+
+    test "batch size is configurable" do
+      enum = build_enumerator(batch_size: 4)
+      products = Product.order(:id).take(4)
+
+      assert_equal([products, products.last.id], enum.first)
+    end
+
+    test "columns are configurable" do
+      enum = build_enumerator(columns: [:updated_at])
+      products = Product.order(:updated_at).take(2)
+
+      expected_product_cursor = products.last.updated_at.strftime(SQL_TIME_FORMAT)
+      assert_equal([products, expected_product_cursor], enum.first)
+    end
+
+    test "columns can be an array" do
+      enum = build_enumerator(columns: [:updated_at, :id])
+      products = Product.order(:updated_at, :id).take(2)
+
+      expected_product_cursor = [products.last.updated_at.strftime(SQL_TIME_FORMAT), products.last.id]
+      assert_equal([products, expected_product_cursor], enum.first)
+    end
+
+    test "columns configured with primary key only queries primary key column once" do
+      queries = track_uncached_queries do
+        enum = build_enumerator(columns: [:updated_at, :id])
+        enum.first
+      end
+      assert_equal 1, queries.first.scan(/`products`.`id`/).count
```

Comment:
This is testing the search for the primary key existing as part of the columns here, right?
https://github.com/Shopify/job-iteration/blob/0090f5de24a26f8470bffe43746c09a11ce522b1/lib/job-iteration/active_record_batch_enumerator.rb#L15
Maybe we could check what the selected columns are instead?
```suggestion
      assert_equal "`products`.`updated_at`, `products`.`id`", queries.first[/SELECT (.*) FROM/, 1]
```